server/s3_common_utilities.h

```diff
@@ -27,6 +27,26 @@
 #include <string>
 
 namespace S3CommonUtilities {
+//Special char hash table
+// '!' -> 33 to '/' -> 47 and ':' -> 58 to '@' -> 64
```

Comment:
M.b. i misunderstood the task...
What is about space char which is encoded as '+'?